dts/bindings/ethernet/nxp,imx-netc-psi.yaml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP i.MX NETC Physical Station Interface (PSI)
+
+compatible: "nxp,imx-netc-psi"
+
+include: [ethernet-controller.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  phy-handle:
+    required: true
+
+  si-index:
+    required: true
+    type: int
+    description: |
+      Station interface index enumerator. Defined by netc_hw_si_idx_t defined in MCUX SDK.
```

Comment:
this is absolutely not allowed in devicetree because it has nothing to do with the hardware.

Reply:
In my opinon, although we defined these information as index, these are exact hardware description for the PSI.
And it's platform related.

It provides information about the PSI's MAC index, VSI/PSI index, and SI index which the hardware designed.
Software has no idea to know these platform hardware info by just PSI base address.

Can we keep it?

Reply:
where is this even being used in the code, anyways, I don't see it ? It gets put in the config struct but I don't see where the value is read from that struct

Reply:
@decsny  Thank you very much. That's the proper way.

Reply:
Too late to the party, but I'd recommend you to do something like we are doing for S32 in order to obtain the sw logical index from the devicetree address: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/ethernet/eth_nxp_s32_netc_priv.h#L91-L96